control-plane/helper/test/test_util.go

```diff
@@ -257,6 +271,27 @@ func SetupK8sAuthMethodWithNamespaces(t *testing.T, consulClient *api.Client, se
 	require.NoError(t, err)
 }
 
+// ResourceHasPersisted checks that a recently written resource exists in the Consul
+// state store with a valid version. This must be true before a resource is overwritten
+// or deleted.
```

Comment:
AFAIK this shouldn't actually be the case for unversioned writes - should we clarify that this is specifically intended to stabilize tests because currently there are issues w/ deletion CAS race conditions? (I'm assuming this is related to [this](https://hashicorp.slack.com/archives/C04AM35BFQT/p1693260271895209) thread)